### ISSUE TITLE
Add feature to support binary data blobs linked with CFE executable

### ISSUE THREAD

**Issue by skliper** [nasa/cFE#326]
In a recent GRC project, CFE was deployed on a virtualized host that had no runtime-accessible persistent storage at all.  Although there was a place to store the CFE executable image itself, the executing runtime image could not read or write to this area as a filesystem.

The apps were statically linked with the executable, but there is no place to put the startup script (cfe_es_startup.scr).

A generic solution to this problem was to add a method for the CMake build to include arbitrary "binary blobs" and link these directly into the executable image, so the PSP/CFE can access them at runtime.  The content of these blobs can be entirely implementation-specific.  The CFE framework just needs an "out of the box" way to link these images into the executable to make deployment on this type of platform easier.

For the GRC use-case, we used this approach to store an "initial ramdisk" tar file which the PSP then untar'ed into the ramdisk before starting CFE.  However the approach could be used for storing any arbitrary data file, such as an FPGA bitfile or DSP/microcontroller firmware.

**Comment by skliper** [comment]
_Imported from trac issue 295.  Created by jphickey on 2019-05-23T10:19:18, last modified: 2019-07-03T13:15:28_

**Comment by skliper** [comment]
_Trac comment by jphickey on 2019-05-23 10:23:00:_

This is already implemented, now isolating into a a change set that can be merged into the framework if the community is interested.

**Comment by skliper** [comment]
_Trac comment by jphickey on 2019-05-24 13:31:08:_

Pushed commit [changeset:5ff9d28] on branch `trac-295-embed-files` for review

**Comment by skliper** [comment]
_Trac comment by jhageman on 2019-06-05 13:10:38:_

CCB 6/5/2019 - code reviewed and approved

**Comment by skliper** [comment]
_Trac comment by jphickey on 2019-06-06 12:03:18:_

Merged into `ic-ccb-2019-06-05` for integration

**Comment by skliper** [comment]
_Trac comment by jhageman on 2019-06-13 09:54:53:_

CCB 6/12/2019 - Approved for integration

**Comment by skliper** [comment]
_Trac comment by jphickey on 2019-06-14 10:05:00:_

Merged into `development` branch

**Comment by skliper** [comment]
_Trac comment by jhageman on 2019-07-03 13:15:28:_

Moved closed tickets targeting 6.6.1 to 6.7.0 end of summer release.
